mautic/mautic
# Fixed issue where campaign's email replies were not recognized

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | Y
| New feature? | 
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
This was hard to reproduce and required just the right HTML. I was only able to reproduce it once but was able to grab the HTML that did and so created a unit test. In some cases, the HTML caused the preg_match to include HTML form the unsubscribe link so the stat wasn't found. This PR thus updates the regex to be more strict so that it only matches the gif's tracking hash ID.

The PR also prevented a PHP notice and set timeouts for all imap functions to prevent locked web requests due to bad ports. 

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. Not sure. I used apple mail to reply to the email. But subsequent attempts to reproduce failed. 

For the timeout issue, I was able to reproduce it by configuring an open non-imap port like 80. I couldn't reload the config after that. 

#### Steps to test this PR:
1. Run the new unit test.
2. Configure the imap settings with a non-imap but open port like 80. After 15 seconds, it should stop trying to connect and load the configuration page.
